packages/core/src/external/commonwell/document/document-file-type-detector.ts

```diff
@@ -0,0 +1,88 @@
+import { Document } from "./document-downloader";
+
+/**
+ * The function `detectFileType` uses magic numbers to determine the file type of a given file.
+ * Magic numbers are unique sequences of bytes that identify the file format or protocol.
+ * In this case, the function reads the first 5 bytes (the magic number) of the file buffer and
+ * matches it against known file type headers.
+ *
+ * @param {Buffer} fileBuffer - The `fileBuffer` parameter is a `Buffer` object that represents the
+ * contents of a file. The first 5 bytes of this buffer, which generally contain the magic number, are used to
+ * identify the file type.
+ * @returns The function `detectFileType` returns a string representing the detected file type.
+ */
+export function detectFileType(fileBuffer: Buffer, document: Document): string {
+  console.log(`Detecting file type for file: ${fileBuffer.slice(0, 10).toString("hex")}`);
```

Comment:
Let's avoid console on functions that can be used in multiple scenarios, please.

If we really want to have this, maybe could add an optional param `log: typeof console.log = () => {}` so it can be enabled only when needed?

Reply:
same for other ones below